week6/assignment2_3/includes/creature.hpp

```diff
@@ -0,0 +1,36 @@
+#ifndef CREATURE_HPP
+#define CREATURE_HPP
+#include <iostream>
+#include <cassert>
+
+// Stores type of an object
+enum class Type {
+    robot,
+    human,
+    alien
+};
+
+struct Creature
+{
+    // Default constructor makes human-type if user don't give any arguments
+    Creature();
+    // Own constructors for different types
+    explicit Creature(const char human_type);
+    explicit Creature(const int robot_type);
+    explicit Creature(const float alien_type);
+    // Stores creature's id
+    union {
+        char human_id;  // first letter of the name
+        int robot_id;  // id-number
+        float alien_id;  // approx wavelength of the light alien's eyes emit
+    };
+    Type type;
+    // To check what type is stored to object
+    std::string check_creature_type();
```

Comment:
Here, you could name this only check_type, as creature is the name of the class, and is implied. 🙂